### PR TITLE
Improve log in baseclient.cpp

### PR DESCRIPTION
**Pull request by huijunw**
1. errno =>  gai_strerror(error)
2. LOG(ERROR) << ... << errno << std::endl; => PLOG(ERROR)

### CODE REVIEW

File: heron/common/src/cpp/network/baseclient.cpp [hunk 1]
```diff
@@ -72,7 +73,8 @@ void BaseClient::Start_Base() {
     struct sockaddr_in t;
     int error = IpUtils::getAddressInfo(t, options_.get_host().c_str(), PF_INET, SOCK_STREAM);
     if (error) {
-      LOG(ERROR) << "getaddrinfo failed in Client " << errno << "\n";
+      LOG(ERROR) << "getaddrinfo failed in Client " << options_.get_host().c_str()
```
Comment (congwang) [inline heron/common/src/cpp/network/baseclient.cpp]: No need c_str() here, operator<< works well with string

File: heron/common/src/cpp/network/baseclient.cpp [hunk 2]
```diff
@@ -72,7 +73,8 @@ void BaseClient::Start_Base() {
...
+          << ": "<< gai_strerror(error) << "\n";
```
Comment (congwang) [inline heron/common/src/cpp/network/baseclient.cpp]: No need newline at the end.

File: heron/common/src/cpp/network/baseclient.cpp (hunk 1 above)
Comment (huijunw) [inline heron/common/src/cpp/network/baseclient.cpp]: done

File: heron/common/src/cpp/network/baseclient.cpp (hunk 2 above)
Comment (huijunw) [inline heron/common/src/cpp/network/baseclient.cpp]: done

**Pull request merged**